src/canvasPanelComponents/slideshow/src/components/Slideshow/SlideShow.tsx

```diff
@@ -0,0 +1,257 @@
+import TapDetector from '../TapDetector/TapDetector';
+import SimpleSlideTransition from '../SimpleSlideTransition/SimpleSlideTransition';
+import ProgressIndicator from '../ProgressIndicator/ProgressIndicator';
+import Slide from '../Slide/Slide';
+// @ts-ignore
+import CanvasNavigation from '../CanvasNavigation/CanvasNavigation.tsx';
+import PeekComponent from '../PeekComponent/PeekComponent';
+
+import './SlideShow.scss';
+
+interface SlideShowProps {
+  mobileBreakPoint: number;
+  backgroundColor: string;
+  addressable: boolean;
+  id: number;
+  jsonLd: object;
+  manifestUri: string;
+  renderPanel: () => void;
+  bem: any;
+}
+
+const SlideShow: React.FC<SlideShowProps> = ({
+  mobileBreakPoint = 767,
+  backgroundColor = '#000000',
+  addressable = false,
+  id,
+  jsonLd,
+  manifestUri,
+  renderPanel,
+  bem,
+}) => {
+  const [innerWidth, setInnerWidth] = useState(0);
+  const [qualifiesForMobile, setQualifiesForMobile] = useState(false);
+  const [isMobileFullScreen, setIsMobileFullScreen] = useState(false);
+  const [down, setDown] = useState(false);
+  const [open, setOpen] = useState(false);
+  const [offset, setOffset] = useState(0);
+  const [inFocus, setInFocus] = useState(false);
+  let touchDetector: any | null = null;
+  let viewport: any;
+  const slideshowEl = useRef(null);
+
+  useEffect(() => {
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    touchDetector = null;
+  }, [down]);
+
+  useEffect(() => {
+    if (touchDetector !== null) {
+      touchDetector.onTap(onTap);
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [touchDetector]);
+
+  useEffect(() => {
+    window.addEventListener('resize', () => setInnerWidth(window.innerWidth));
+    return () =>
+      window.removeEventListener('resize', () =>
+        setInnerWidth(window.innerWidth)
+      );
+  }, []);
+
+  useEffect(() => {
+    setQualifiesForMobile(window.innerWidth <= mobileBreakPoint);
+  }, [innerWidth, mobileBreakPoint]);
+
+  const nextInRange = (fromHOC: () => void) => {
+    if (viewport) viewport.viewer.viewer.viewport.applyConstraints(true);
+    fromHOC();
+  };
+
+  const previousInRange = (fromHOC: () => void) => {
+    if (viewport) viewport.viewer.viewer.viewport.applyConstraints(true);
+    fromHOC();
+  };
+
+  const onDragStart = () => {
+    setDown(true);
+  };
+
+  const onDragStop = () => {
+    setDown(false);
+    setOffset(0);
+  };
+
+  const applyOffset = (offsetValue: number) => {
+    setOffset(offsetValue);
+  };
+
+  const onOpen = () => {
+    setOpen(true);
+  };
+
+  const onClose = () => {
+    setOpen(false);
+  };
+
+  const onExitFullscreen = () => {
+    setIsMobileFullScreen(false);
+  };
+
+  const onTap = () => {
+    setOpen(!open);
+  };
+
+  const setView = (view: any) => {
+    if (touchDetector && touchDetector !== null) {
+      touchDetector.unbind();
+    }
+    touchDetector = new TapDetector(view.viewer.viewer.canvas);
+    viewport = view;
+  };
```

Comment:
If this was an effect, you could let React do the unbind for you:
```javascript
const [view, setView] = useState();

useLayoutEffect(() => {
  if (view) {
    touchDetector.current = new TapDetector(view);
  }

  return () => { 
    if (touchDetector.current) {
      touchDetector.current.unbind(); 
    }
  }
})
```

Reply:
I think one of my other comments got lost. For the `touchDetector` we could use `useRef` to persist the touch detector between renders.
```
const touchDetector = useRef();
```

and then access/set it using `touchDetector.current`.

Reply:
@stephenwf this was really helpful! Thank you !